src/analysis/liouville/liouville_constant.lean

```diff
@@ -76,6 +74,116 @@ lemma liouville_number_eq_initial_terms_add_tail (hm : 1 < m) (k : ℕ) :
+-- two useful inequalities
+have m0 : 0 < m := (zero_lt_one.trans m1),
+have mi : abs (1 / m) < 1 :=
+  (le_of_eq (abs_of_pos (one_div_pos.mpr m0))).trans_lt ((div_lt_one m0).mpr m1),
+calc (∑' i, 1 / m ^ (i + (n + 1))!)
+    < ∑' i, 1 / m ^ (i + (n + 1)!) :
+    -- to show the strict inequality between these series, we prove that:
+    tsum_lt_tsum_of_nonneg
+      -- 1. the first series has non-negative terms
+      (λ b, one_div_nonneg.mpr (pow_nonneg m0.le _))
+      -- 2. the second series dominates the first
+      (λ b, one_div_pow_le_one_div_pow_of_le m1.le (b.add_factorial_succ_le_factorial_add_succ n))
+      -- 3. the term with index `i = 2` of the first series is strictly smaller than
+      -- the corresponding term of the second series
+      (one_div_pow_strict_mono m1 (n.add_factorial_succ_lt_factorial_add_succ rfl.le))
+      -- 4. the second series is summable, since its terms grow quickly
+      (summable_one_div_pow_of_le m1 (λ j, nat.le.intro rfl))
+... = ∑' i, (1 / m) ^ i * (1 / m ^ (n + 1)!) :
+    -- split the sum in the exponent and massage
+    by { congr, ext i, rw [pow_add, ← div_div_eq_div_mul, div_eq_mul_one_div, ← one_div_pow i] }
+-- factor the constant `(1 / m ^ (n + 1)!)` out of the series
+... = (∑' i, (1 / m) ^ i) * (1 / m ^ (n + 1)!) : tsum_mul_right
+... = (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) :
+    -- the series if the geometric series
+    mul_eq_mul_right_iff.mpr (or.inl (tsum_geometric_of_abs_lt_1 mi))
+
+lemma aux_calc (n : ℕ) (hm : 2 ≤ m) :
+  (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) ≤ 1 / (m ^ n!) ^ n :=
+calc (1 - 1 / m)⁻¹ * (1 / m ^ (n + 1)!) ≤ 2 * (1 / m ^ (n + 1)!) :
+  -- the second factors coincide (and are non-negative),
+  -- the first factors, satisfy the inequality `sub_one_div_inv_le_two`
+  mul_mono_nonneg (one_div_nonneg.mpr (pow_nonneg (zero_le_two.trans hm) _))
+    (sub_one_div_inv_le_two hm)
+... = 2 / m ^ (n + 1)! : mul_one_div 2 _
+... = 2 / m ^ (n! * (n + 1)) : congr_arg ((/) 2) (congr_arg (pow m) (mul_comm _ _))
+... ≤ 1 / m ^ (n! * n) :
+  begin
+    -- [ NB: in this block, I do not follow the brace convention for subgoals -- I wait until
+    --   I solve all extraneous goals at once with `exact pow_pos (zero_lt_two.trans_le hm) _`. ]
+    -- Clear denominators and massage*
+    apply (div_le_div_iff _ _).mpr,
+    conv_rhs { rw [one_mul, mul_add, pow_add, mul_one, pow_mul, mul_comm, ← pow_mul] },
+    -- the second factors coincide, so we prove the inequality of the first factors*
+    apply (mul_le_mul_right _).mpr,
+    -- solve all the inequalities `0 < m ^ ??`
+    any_goals { exact pow_pos (zero_lt_two.trans_le hm) _ },
+    -- `2 ≤ m ^ n!` is a consequence of monotonicity of exponentiation at `2 ≤ m`.
+    exact trans (trans hm (pow_one _).symm.le) (pow_mono (one_le_two.trans hm) n.factorial_pos)
+  end
+... = 1 / (m ^ n!) ^ n : congr_arg ((/) 1) (pow_mul m n! n)
+
+end two_useful_inequalities
 
 end m_is_real
+
+section m_is_natural
```

Comment:
You are probably right: I had them in since I wanted to stress that the variable `m` started out being any real number, but became a natural number at some point.  Now the file is so short that maybe it is no longer so important to make this so obvious.  I am happy to remove the sections, if you think that it helps.  Maybe, I could write a doc-string, saying that we begin with proving results about general real numbers and specialize to naturals later on.

Reply:
Yeah, maybe a small comment in between the lemmata. More doc-strings can never hurt anyone!

Reply:
I just added it and expanded slightly the documentation